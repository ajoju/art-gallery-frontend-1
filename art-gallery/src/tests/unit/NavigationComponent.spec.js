import { mount } from "@vue/test-utils";
import { expect, it, describe, beforeEach } from "vitest";
import CardComponent from "@/components/CardComponent.vue";

describe("CardComponent", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(CardComponent, {
      propsData: {
        heading: "Card Heading",
        imageURL: "https://example.com/image.jpg",
        subheading: "Card Subheading",
        detail1: "Detail 1",
        detail2: "Detail 2",
        detail3: "Detail 3",
        detail4: "Detail 4",
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("displays the correct heading", () => {
    const heading = wrapper.find(".card-heading");
    expect(heading.text()).toBe("Card Heading");
  });

  it("displays the correct image URL", () => {
    const image = wrapper.find(".card-img");
    expect(image.attributes("src")).toBe("https://example.com/image.jpg");
  });

  it("displays the correct subheading", () => {
    const subheading = wrapper.find(".card-subheading");
    expect(subheading.text()).toBe("Card Subheading");
  });

  it("displays the correct details", () => {
    const details = wrapper.findAll(".detail");
    expect(details.length).toBe(3);
    expect(details[0].text()).toBe("Detail 1");
    expect(details[1].text()).toBe("Detail 2");
    expect(details[2].text()).toBe("Detail 3");
  });

  it("displays the correct detail description", () => {
    const detailDescription = wrapper.find(".detail-description");
    expect(detailDescription.text()).toBe("Detail 4");
  });
});
