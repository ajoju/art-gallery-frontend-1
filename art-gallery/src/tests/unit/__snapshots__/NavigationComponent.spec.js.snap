// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardComponent > renders correctly 1`] = `
"<div data-v-f967684d=\\"\\" class=\\"card-div\\">
  <h1 data-v-f967684d=\\"\\" class=\\"card-heading\\">Card Heading</h1><img data-v-f967684d=\\"\\" class=\\"card-img\\" src=\\"https://example.com/image.jpg\\">
  <h2 data-v-f967684d=\\"\\" class=\\"card-subheading\\">Card Subheading</h2>
  <h4 data-v-f967684d=\\"\\" class=\\"detail\\">Detail 1</h4>
  <h4 data-v-f967684d=\\"\\" class=\\"detail\\">Detail 2</h4>
  <h4 data-v-f967684d=\\"\\" class=\\"detail\\">Detail 3</h4>
  <h4 data-v-f967684d=\\"\\" class=\\"detail-description\\">Detail 4</h4>
</div>"
`;
